### PR TITLE
84372: Add DR scheduled job FailureNotificationEmailJob 

### PR DESCRIPTION
**Pull request by dfong-adh**
## Summary
Adds a scheduled job `FailureNotificationEmailJob` to send an email to the Veteran if their submitted form or evidence attachment errored during processing.

This feature is controlled by the flipper flag `decision_review_failure_notification_email_job_enabled`. The job is scheduled to run once a day.

This is owned by the Benefits Decision Reviews team.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/84372

## Testing done
- [x] *New code is covered by unit tests*
This has been tested locally and with rspec tests
- *If this work is behind a flipper:*
Flipper flag will be used to test in staging before enabling in production.

## What areas of the site does it impact?
This PR impacts the `appeal_submissions` and `appeal_submission_uploads` tables as we are reading and writing to those tables to check if an email has already been sent to the Veteran.

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/models/appeal_submission_upload.rb [hunk 1]
```diff
@@ -9,4 +9,16 @@ class AppealSubmissionUpload < ApplicationRecord
...
+  scope :failure_not_sent, -> { where(failure_notification_sent_at: nil).order(id: :asc) }
+
+  # Matches all characters except first 3 characters, last 6 characters (2 + .pdf extension), underscores, and hyphens
+  MASK_REGEX = /(?<=.{3})[^_-](?=.{6})/
+
+  def masked_attachment_filename
+    filename = JSON.parse(decision_review_evidence_attachment&.file_data || '{}')['filename']
+    raise 'Filename for not found' if filename.nil?
+
+    filename.gsub(MASK_REGEX, '*')
```
Comment (shaunburdick) [inline app/models/appeal_submission_upload.rb:22]: I think this should be `X` and not `*` as the VA Notify template uses Markdown and `*` is a special character in Markdown

**Review by stevenjcumming — APPROVED** [review]

File: app/sidekiq/decision_review/failure_notification_email_job.rb [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        record_evidence_email_send_successful(upload)
+      rescue => e
+        record_evidence_email_send_failure(upload, e)
+      end
+    end
+
+    def record_form_email_send_successful(submission)
+      metadata = { submitted_appeal_uuid: submission.submitted_appeal_uuid, form_type: submission.type_of_appeal }
+      Rails.logger.info('DecisionReview::FailureNotificationEmailJob form email queued', metadata)
+      StatsD.increment("#{STATSD_KEY_PREFIX}.form.email_queued", tags: ["form_type:#{submission.type_of_appeal}"])
```
Comment (kayline) [inline app/sidekiq/decision_review/failure_notification_email_job.rb:107]: When do we set the `delete_date` on the SavedClaim? Also, where does `failure_notification_sent_at` get set on the AppealSubmission?

File: app/sidekiq/decision_review/failure_notification_email_job.rb [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      end
+    end
+
+    def submission_uploads
+      @submission_uploads ||= begin
+        uploads = errored_saved_claims.map { |sc| JSON.parse(sc.metadata)['uploads'] }
+        ids = uploads.flatten.select { |upload| upload&.fetch('status') == ERROR_STATUS }.pluck('id')
+
+        ::AppealSubmissionUpload.where(lighthouse_upload_id: ids).failure_not_sent
+      end
```
Comment (kayline) [inline app/sidekiq/decision_review/failure_notification_email_job.rb:63]: Is this going to catch evidence uploads where the claim was successful but the evidence failed?

**Review by kayline — COMMENTED** [review]
Added a couple questions about tracking

File: app/sidekiq/decision_review/failure_notification_email_job.rb (hunk 2 above)
Comment (dfong-adh) [reply app/sidekiq/decision_review/failure_notification_email_job.rb:107]: `delete_date` is set in other Sidekiq jobs (the SavedClaim status updater jobs [here](https://github.com/department-of-veterans-affairs/vets-api/tree/master/app/sidekiq/decision_review)). 

`failure_notification_sent_at` is set on line 83.

**Review by kayline — APPROVED** [review]
Thanks, makes sense!

**Pull request merged**